### PR TITLE
DIA-4059 refactor android interface

### PR DESCRIPTION
**Pull request by Nevazhnovu**

### CODE REVIEW

File: Assets/ConsentManagementProvider/Scripts/facade/CMP.cs [hunk 1]
```diff
@@ -26,13 +26,12 @@ internal static ISpSdk ConcreteInstance
...
-                    concreteInstance = new ConsentWrapperAndroid();
-#elif UNITY_IOS && !UNITY_EDITOR_OSX
-                    concreteInstance = new ConsentWrapperIOS();
-#else
-                    concreteInstance = new ConsentWrapperUnityEditor();
-#endif
+                    concreteInstance = Application.platform switch
+                    {
+                        RuntimePlatform.Android => new ConsentWrapperAndroid(),
+                        RuntimePlatform.IPhonePlayer => new ConsentWrapperAndroid(),
```
Comment (andresilveirah) [inline Assets/ConsentManagementProvider/Scripts/facade/CMP.cs:32]: > RuntimePlatform.IPhonePlayer => new ConsentWrapperAndroid()
@Nevazhnovu is this line correct?

File: Assets/ConsentManagementProvider/Scripts/json/wrappers/GdprConsentWrapper.cs [hunk 2]
```diff
@@ -9,11 +9,7 @@ internal class GdprConsentWrapper
         public Dictionary<string, object> TCData;
         public Dictionary<string, SpVendorGrantWrapper> grants;
         public List<string> acceptedCategories;
-        public bool applies;
         public string webConsentPayload;
         public ConsentStatusWrapper consentStatus;
-#nullable enable
-        public GCMDataWrapper? gcmStatus;
-#nullable disable
     }
```
Comment (andresilveirah) [inline Assets/ConsentManagementProvider/Scripts/json/wrappers/GdprConsentWrapper.cs:14]: @Nevazhnovu were `applies` an `gcmStatus` removed on purpose?

File: Assets/ConsentManagementProvider/Scripts/facade/CMP.cs (hunk 1 above)
Comment (Nevazhnovu) [reply Assets/ConsentManagementProvider/Scripts/facade/CMP.cs:32]: @andresilveirah you are right, that's a typo. My bad!

File: Assets/ConsentManagementProvider/Scripts/json/wrappers/GdprConsentWrapper.cs (hunk 2 above)
Comment (Nevazhnovu) [reply Assets/ConsentManagementProvider/Scripts/json/wrappers/GdprConsentWrapper.cs:14]: @andresilveirah Yup, that's done purposefully. The `applies` and `gcmStatus` have moved to `SpGdprConsentWrapper<platform>.cs` classes

**Pull request merged**